lib/version.sh

```diff
@@ -5,7 +5,7 @@ set -Eeuo pipefail
 source lib/fetch.sh
 
 ARTIFACTS_DOMAIN=${ARTIFACTS_DOMAIN:="https://artifacts.aerospike.com"}
-RE_VERSION='[0-9]+[.][0-9]+[.][0-9]+([.][0-9]+)+(-rc[0-9]+)*'
+RE_VERSION='[0-9]+[.][0-9]+[.][0-9]+([.][0-9]+)+(-rc[0-9]+)*+([-][0-9]*[-]g[0-9a-zA-Z]*)*'
```

Comment:
That is getting a bit long - also what is the `+` that follows the `*` after the rc parsing doing? Many of your `*` can be replaced with `?` for 0 or 1 times instead of 0 or more.

I haven't tested the following:
```
[0-9]+([.][0-9]){3}(-rc[0-9]+)?([-][0-9]*[-]g[0-9a-zA-Z]*)?
```